src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -557,31 +536,31 @@ def select(self, cols: List[Expression]) -> "SelectStatement":
         return new
 
     def filter(self, col: Expression) -> "SelectStatement":
-        if self.flatten_disabled:
-            can_be_flattened = False
-        else:
-            dependent_columns = derive_dependent_columns(col)
-            can_be_flattened = can_clause_dependent_columns_flatten(
-                dependent_columns, self.column_states
-            )
+        dependent_columns = derive_dependent_columns(col)
```

Comment:
Similarly, moving this to the top has extra calculation when `self.flatten_disabled is True`.